contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -900,8 +900,34 @@ clearly described and the permissions issues should be resolved.
 This could be done with a double opt-in (an opt-in from both the referrer and the refer-ee) or with secondary permissions
 checks performed in admission. 
 
-TODO: Plugins, extensions, nested kinds, headers
+TODO: Plugins, extensions, headers
 
+### Handling references to multiple kinds
+
+References which can refer to multiple kinds should use a single field, and select the target kind via `apiVersion` and `kind` fields.
+
+For example, if one can retrieving a referenced value from a `ConfigMap` or a `Secret` kind, the schema for the reference should be of the form:
+
+```yaml
+# preferred pattern
+valueFrom:
+  kind: Secret  # alternatively ConfigMap
```

Comment:
A kind is a schema. A resource is a path in the API. You need the latter if you want to be certain to uniquely identify a particular object in the general case. (mentioned in previous comment thread but repeating here just for clarity)

Reply:
can you explain why a kind paired with a group wouldn't uniquely identify a resource? [my thoughts are in this comment](https://github.com/kubernetes/community/pull/5748/files#r627573670), but in summary I believe the identifier is unique whether it's kind or resource (AKA plural as it's called in the CRD).

Reply:
Because a Kind can be mounted in the API in more than one place. The only example today is Scale.

You need a resource to make an API request, because the resource is in the URL path. You cannot make an API request if all you know is the Kind. So, in addition to future-proof-ness, using a resource means less work for clients.

Reply:
> Because a Kind can be mounted in the API in more than one place. The only example today is Scale.

David was mentioning he's seen a lot of other examples in openshift.

Reply:
> David was mentioning he's seen a lot of other examples in openshift.

Further examples up here https://github.com/kubernetes/community/pull/5748/files#r628142544

Reply:
PR now recommends resource instead of kind.